specify/specify7
# Implement Exchange Out, Disposal, Gift, and Deaccession Virtual Calculated Fields

Fixes #2521

Implements the calculations done in Specify 6 into Specify 7 as virtual fields for the following:
Exchange Out:
- Total Preps (Sum of all associated exchange out preparations)
- Total Items (Sum of all associated collection object preparations)

Disposal:
- Total Preps (Sum of all associated disposal preparations)
- Total Items (Sum of all associated collection object preparations)

Gift:
- Total Preps (Sum of all associated gift preparations)
- Total Items (Sum of all associated collection object preparations)

Deaccession’s
- Total Preps (Sum of all associated disposal, gift, and exchange out preparations)
- Total Items (Sum of all associated collection object preparations)

Note: This PR is also taking the opportunity to take the existing hard-coded sql code in the `calculated_fields.py` file, and re-implementing it using the Django ORM.

### Checklist

      and self-explanatory (or properly documented)

### Testing instructions

- Go to the forms of ExchangeOut, Disposal, Gift, and Deaccession.  
- Play around with adding and deleting preparations
- Ensure Total Preps and Total Items display accurate values
